lib/paranoia.rb

```diff
@@ -2,6 +2,7 @@
 
 module Paranoia
   @@default_sentinel_value = nil
+  @@default_recovery_window = nil
```

Comment:
Does `lib/paranoia.rb` ever get run in a thread?  If so, the use of class variables will not work.  Each thread can independently set the class variable, but they all share the same class variable.

In a multithreaded environment, such as in sidekiq, use thread-local variables:

```
    Thread.current[:default_sentinel_value] = nil
    Thread.current[:default_recovery_window] = nil
```

Reply:
Hmm, I understand this concern. My thoughts:

1. This is following convention in the existing gem (class var getter/setter) https://github.com/rubysherpas/paranoia/blob/core/lib/paranoia.rb

2. We are hoping to get this merged upstream (hence wanting to follow their convention)

3. The class variables are only ever set in a rails initializer.

In practice these variables are more than thread safe — they are just "hooks" to give gem users the ability to set constants for the gem once, when the app loads.

But in theory, you are 100% right, it is not thread-safe if someone chose to dynamically change these values. I personally can't imagine it ever being an issue. If you feel strongly about it we could re-strategize about what to do re: upstream.